src/test/java/teammates/test/cases/action/StudentFeedbackResultsPageActionTest.java

```diff
@@ -213,4 +226,23 @@ public void testExecuteAndPostProcess() throws Exception {
     protected StudentFeedbackResultsPageAction getAction(String... params) {
         return (StudentFeedbackResultsPageAction) gaeSimulation.getActionObject(getActionUri(), params);
     }
+
+    @Override
+    @Test
+    protected void testAccessControl() throws Exception {
+        FeedbackSessionAttributes session1InCourse1 = dataBundle.feedbackSessions
+                .get("session1InCourse1");
+        FeedbackSessionsLogic.inst().publishFeedbackSession(session1InCourse1);
+
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.COURSE_ID, session1InCourse1.getCourseId(),
+                Const.ParamsNames.FEEDBACK_SESSION_NAME,
+                session1InCourse1.getFeedbackSessionName()
+        };
+
+        verifyOnlyStudentsOfTheSameCourseCanAccess(submissionParams);
+
+        // TODO: test no questions -> redirect after moving detection logic to
```

Comment:
You may want to investigate this TODO and see if this is still relevant and remove it if it is not.